packages/design-system/src/styles/components/_Button.scss

```diff
@@ -89,208 +78,195 @@
+    border-color-disabled: $color-gray-lighter,
+  ),
+  button--danger: (
+    text-decoration: $button-text-decoration-none,
+    bg-color: $color-error,
+    text-color: $color-white,
+    border-color: $color-transparent,
+    bg-color-hover: $color-error-dark,
+    text-color-hover: $color-white,
+    border-color-hover: $color-transparent,
+    bg-color-active: $color-error-darker,
+    text-color-active: $color-white,
+    border-color-active: $color-transparent,
+    bg-color-disabled: $color-gray-lighter,
+    text-color-disabled: $color-gray-dark,
+    border-color-disabled: $color-gray-lighter,
+  ),
+  button--transparent: (
+    text-decoration: $button-text-decoration-underline,
+    bg-color: $button-transparent-bg,
+    text-color: $button-transparent-text-color,
+    border-color: $button-transparent-border-color,
+    bg-color-hover: $button-transparent-bg--hover,
+    text-color-hover: $button-transparent-text-color--hover,
+    border-color-hover: $button-transparent-border-color--hover,
+    bg-color-active: $button-transparent-bg--active,
+    text-color-active: $button-transparent-text-color--active,
+    border-color-active: $button-transparent-border-color--active,
+    bg-color-disabled: $button-transparent-bg--disabled,
+    text-color-disabled: $button-transparent-text-color--disabled,
+    border-color-disabled: $button-transparent-border-color--disabled,
+  ),
+  button--inverse: (
+    text-decoration: $button-text-decoration-none,
+    bg-color: $button-inverse-bg,
+    text-color: $button-inverse-text-color,
+    border-color: $button-inverse-border-color,
+    bg-color-hover: $button-inverse-bg--hover,
+    text-color-hover: $button-inverse-text-color--hover,
+    border-color-hover: $button-inverse-border-color--hover,
+    bg-color-active: $button-inverse-bg--active,
+    text-color-active: $button-inverse-text-color--active,
+    border-color-active: $button-inverse-border-color--active,
+    bg-color-disabled: $button-inverse-bg--disabled,
+    text-color-disabled: $button-inverse-text-color--disabled,
+    border-color-disabled: $button-inverse-border-color--disabled,
+  ),
 );
 
-@each $colors in map-keys($button-color-variants) {
-  $color: map-get($button-color-variants, $colors);
-
-  .ds-c-button--#{$colors},
-  .ds-c-button--#{$colors}:visited {
-    background-color: nth($color, 1);
-    border-color: transparent;
-    color: $color-white;
+@each $label, $map in $buttons-map {
+  .ds-c-#{$label},
+  .ds-c-#{$label}:visited {
+    background-color: map-get($map, bg-color);
+    border-color: map-get($map, border-color);
+    color: map-get($map, text-color);
+    text-decoration: map-get($map, text-decoration);
 
     &:hover,
     &:focus,
     &:active,
     &.ds-c-button--hover,
     &.ds-c-button--focus,
     &.ds-c-button--active {
-      border-color: transparent;
-      color: $color-white;
+      border-color: map-get($map, border-color);
+      color: map-get($map, text-color);
```

Comment:
[question] what is the point of this rule section? It seems like the `border-color` & `color` are all getting overwritten for specific states in the subsequent rules, so when does this ruleset ever get used?

Reply:
Thanks and you're right - I'll removed this section